flexget/plugins/filter/movie_queue.py

```diff
@@ -396,3 +432,275 @@ def queue_get(session=None, downloaded=False):
+        'quality': {'type': 'string', 'format': 'quality_requirements'},
+        'reset_downloaded': {'type': 'boolean', 'default': False}
+    },
+    'anyOf': [
+        {'required': ['quality']},
+        {'required': ['reset_downloaded']}
+    ]
+}
+
+movie_edit_results_schema = api.schema('movie_edit_results_schema', movie_edit_results_schema)
+movie_edit_input_schema = api.schema('movie_edit_input_schema', movie_edit_input_schema)
+
+
+@movie_queue_api.route('/')
+class MovieQueueAPI(APIResource):
+    @api.response(404, 'Page does not exist')
+    @api.response(200, 'Movie queue retrieved successfully', movie_queue_schema)
+    @api.doc(parser=movie_queue_parser)
+    def get(self, session=None):
+        """ List queued movies """
+        args = movie_queue_parser.parse_args()
+        page = args['page']
+        max_results = args['max']
+        downloaded = args['downloaded']
+        sort_by = args['sort_by']
+        order = args['order']
+
+        movie_queue = queue_get(session=session, downloaded=downloaded)
+        count = len(movie_queue)
+
+        pages = int(ceil(count / float(max_results)))
+
+        movie_items = []
+
+        if page > pages and pages != 0:
+            return {'error': 'page %s does not exist' % page}, 404
+
+        start = (page - 1) * max_results
+        finish = start + max_results
+        if finish > count:
+            finish = count
+
+        for movie_number in range(start, finish):
+            movie_items.append(movie_queue[movie_number].to_dict())
+
+        sorted_movie_list = sorted(movie_items, key=itemgetter(sort_by), reverse=order)
+
+        return jsonify({
+            'movies': sorted_movie_list,
+            'number_of_movies': count,
+            'page_number': page,
+            'total_number_of_pages': pages
+        })
+
+    @api.response(400, 'Page not found')
+    @api.response(200, 'Movie successfully added', movie_add_results_schema)
+    @api.validate(movie_add_input_schema)
+    def post(self, session=None):
+        """ Add movies to movie queue """
+        kwargs = request.json
+
+        try:
+            kwargs['quality'] = qualities.Requirements(kwargs.get('quality'))
+        except ValueError as e:
+            reply = {
+                'status': 'error',
+                'message': e.message
+            }
+            return reply, 400
+        kwargs['session'] = session
+
+        try:
+            movie = queue_add(**kwargs)
+        except QueueError as e:
+            reply = {
+                'status': 'error',
+                'message': e.message
+            }
+            return reply, 500
+
+        return jsonify(
+            {
+                'message': 'Successfully added movie to movie queue',
+                'movie': movie
+            }
+        )
+
+
+@movie_queue_api.route('/<int:movie_id>/')
+@api.doc(params={'movie_id': 'ID of Queued Movie'})
+class MovieQueueManageAPI(APIResource):
+    @api.response(400, 'Page not found')
```

Comment:
400 is never returned by this api so this should be removed.

500 is returned though so maybe it should be changed to 500 with a description of some sort